napari/_qt/widgets/qt_keyboard_settings.py

```diff
@@ -400,43 +430,57 @@ def _set_keybinding(self, row, col):
             # get the current action name
             current_action = self._table.item(row, self._action_col).text()
 
-            # get the original shortcutS
-            current_shortcuts = list(
-                action_manager._shortcuts.get(current_action, [])
-            )
-            for mod in {"Shift", "Ctrl", "Alt", "Cmd", "Super", 'Meta'}:
-                # we want to prevent multiple modifiers but still allow single modifiers.
-                if new_shortcut.endswith('-' + mod):
-                    self._show_bind_shortcut_error(
-                        current_action,
-                        current_shortcuts,
-                        row,
-                        new_shortcut,
-                    )
-                    return
+            # get the original shortcuts
+            current_shortcuts = self._find_shortcuts(current_action)
 
-            # Flag to indicate whether to set the new shortcut.
-            replace = self._mark_conflicts(new_shortcut, row)
+            # FIXME change conflict detection method to match new system
+            # replace = self._mark_conflicts(new_shortcut, row)
+            replace = True
```

Comment:
This looks like unfinished as make following if obsolete.

Reply:
I decided not to fix this because a new GUI/conflict determination method will be need to be made when @dalthviz takes over the front-end side of things. there's also a non-breaking but confusing bug where settings will sometimes appear in the wrong order so this code should not be released until his work is done as well

Reply:
Hi there, I did a quick check on this and just to see if I'm getting the idea of the key bindings system correctly, probably in a new implementation of the `_mark_conflicts` method, it will use some of the key bindings util functions like the `create_conflict_filter` one, right? Maybe even following the NAP a `has_conflicts` util function will need to be created, right?

Reply:
yes, there is a `has_conflicts` function under `_dispatch.py`

Reply:
Checking this a little bit more, if the usage of marking conflicts should be preserved here (even if the logic is not fully migrated to use the new keybinding utils), we could define `_mark_conflicts` like:

```python
    def _mark_conflicts(self, new_shortcut, row) -> bool:
        # Go through all layer actions to determine if the new shortcut is already here.
        current_action = self._table.item(row, self._action_col).text()
        actions_all = self._get_layer_actions()
        current_item = self._table.currentItem()
        for row1, (action_name, _action) in enumerate(actions_all.items()):
            shortcuts = [kb2str(shortcut.keybinding) for shortcut in self._find_shortcuts(action_name)]
            if kb2str(new_shortcut) not in shortcuts:
                continue
            # Shortcut is here (either same action or not), don't replace in settings.
            if action_name != current_action:
                # the shortcut is saved to a different action

                # show warning symbols
                self._show_warning_icons([row, row1])

                # show warning message
                message = trans._(
                    "The keybinding <b>{new_shortcut}</b> is already assigned to <b>{action_description}</b> and may conflict with it.",
                    new_shortcut=new_shortcut,
                    action_description=self._app.commands[action_name].title,
                )
                self._show_warning(row, message)

                self._restore_shortcuts(row)

                self._cleanup_warning_icons([row, row1])

                return False

            # This shortcut was here.  Reformat and reset text.
            format_shortcut = Shortcut(new_shortcut).platform
            with lock_keybind_update(self):
                current_item.setText(format_shortcut)

        return True
```

So leaving the logic as it is but using the `kb2str` util when comparing shortcuts/keybindings